processing/src/test/java/org/apache/druid/segment/serde/cell/CellWriterTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.segment.serde.cell;
+
+import org.apache.druid.segment.writeout.OnHeapMemorySegmentWriteOutMedium;
+
+public class CellWriterTest extends BytesReadWriteTestBase
```

Comment:
This is a super meta comment, this PR is introducing classes `CellWriter` and `CellReader` that are intending to be reusable.  There is a `CellWriter` test, but no `CellReader` test and I haven't dug in yet to validate that there is some other test which uses both of them at the same time.  Sometimes, when people are trying to learn how to use a thing that is intended to be reusable, they look for some test classes that can show them the usage pattern.  At least, right now, I do not know which test classes to point someone at to understand how to use `CellWriter` and `CellReader`.

Even for the `CellWriterTest`, it is currently leveraging the `TestBase` that really helps with validating the technical correctness, but the abstraction doesn't make it easy to see how to use the thing as a black box.  I made other comments asking for class-level javadoc, I'd like to also suggest some test somewhere whose goal is more to show how to use the thing and less actually validating technical correctness (of course, it should validate correctness too, just, the primary intent of the test should be an example of how to use the APIs).  The class-level javadoc can refer to that test to help solidify how teh thing should be used as well.  This suggestion is not saying that the current tests should be changed, it is suggesting the addition of a test somewhere that exemplifies the intended usage pattern.

Reply:
the CellWriter test tests both read and writer. it should be named as such. The same for BlockCompressPayloadWriter test...

they cycle through a set of input byte[], and verify they can read the same bytes out in various cases, and test for sizes as well.

the idioms for how to use the classes are in ByteWriterTestHelper and PayloadWriterValidationFunctionBuilder and CellReaderValidationFunctionBuilder. 

let me think about how to do this. the whole point of the base test was to avoid the fact that both the CellReader/Writer and BlockCompressedPayloadReader/Writer both read/write byte[]. The latter just requires you to manage the "index" of positions yourself. The two classes' APIs are so similar that leveraging the base was essential

i could have a test file for each that showed a single use case perhaps?

discuss in sync, I see your point, though.

Reply:
actually, in the extant teste class CellWriterReaderTest, I can add a single test method that shows how to use classes.